plugins/parsers/json/parser.go

```diff
@@ -51,10 +59,28 @@ func (p *JSONParser) parseObject(metrics []telegraf.Metric, jsonOut map[string]i
 		return nil, err
 	}
 
-	tags, nFields := p.switchFieldToTag(tags, f.Fields)
+	//if time key is specified, set it to time
+	nTime := time.Now().UTC()
+	if p.JSONTimeKey != "" {
+		if p.JSONTimeFormat == "" {
+			err := fmt.Errorf("E! If 'json_time_key' is specified, there must be a 'json_time_format'")
+			return nil, err
+		}
 
-	metric, err := metric.New(p.MetricName, tags, nFields, time.Now().UTC())
+		if f.Fields[p.JSONTimeKey] == nil {
+			err := fmt.Errorf("E! JSON time key could not be found")
+			return nil, err
+		}
+
+		nTime, err = time.Parse(p.JSONTimeFormat, f.Fields[p.JSONTimeKey].(string))
```

Comment:
Make sure to test the type assertion since it comes from input and could be any type.